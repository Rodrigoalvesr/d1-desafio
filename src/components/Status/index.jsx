import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import Icon from '@Components/Icon';
import { makeStyles } from '@material-ui/core/styles';
import getStatus from '@/common/getStatus';
import classnames from 'classnames';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    alignItems: 'center',
  },
  icon: {
    width: 16,
    height: 16,
    marginRight: 10,
  },
  name: {
    fontSize: 13,
  },
});

function Status({ id }) {
  const classes = useStyles();
  const status = React.useMemo(() => getStatus(id), [id]);

  return (
    <div className={classes.root}>
      <Icon className={classes.icon} name={status.icon} color={status.color} />
      <Typography className={classnames(classes.name)}>
        {status.name}
      </Typography>
    </div>
  );
}

Status.propTypes = {
  id: PropTypes.number.isRequired,
};

export default Status;
