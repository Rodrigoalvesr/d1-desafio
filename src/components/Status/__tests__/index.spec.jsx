import React from 'react';
import renderer from 'react-test-renderer';
import Status from '../index';

describe('Status', () => {
  it('Renders correctly', () => {
    const component = renderer.create(<Status id={0} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
