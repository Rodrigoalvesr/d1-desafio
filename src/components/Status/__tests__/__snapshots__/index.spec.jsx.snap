// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <svg
    className="makeStyles-root-4 makeStyles-root-5 makeStyles-icon-2"
  >
    table.svg
  </svg>
  <p
    className="MuiTypography-root makeStyles-name-3 MuiTypography-body1"
  >
    Todos
  </p>
</div>
`;
